lambda.tf

```diff
@@ -1,59 +1,18 @@
 locals {
   source_code_repo_dir_path = "geff"
-  lambda_code_dir           = "lambda_src"
-  output_dist_file_name     = "lambda-code.zip"
   runtime                   = "python3.8"
-  source_code_dist_dir_path = "lambda-code-dist"
-  upload_dir                = "geff"
 }
 
-resource "null_resource" "install_python_dependencies" {
-  # If this always runs archive_file is fine, else we have an issue during refresh:
-  # https://github.com/hashicorp/terraform-provider-archive/issues/78
-  triggers = {
-    always_run = timestamp()
-  }
-
-  provisioner "local-exec" {
-    command = "bash ${path.module}/scripts/create_dist_pkg.sh"
-
-    environment = {
-      source_code_repo_dir_path = "${local.source_code_repo_dir_path}"
-      lambda_code_dir_path      = "${local.source_code_repo_dir_path}/${local.lambda_code_dir}"
-      source_code_dist_dir_path = local.source_code_dist_dir_path
-      source_code_upload_dir    = local.upload_dir
-      runtime                   = local.runtime
-      path_module               = path.module
-      path_cwd                  = path.cwd
-    }
-  }
-}
-
-data "archive_file" "lambda_code" {
-  source_dir  = "${path.module}/${local.source_code_dist_dir_path}/"
-  output_path = "${path.module}/${local.output_dist_file_name}"
 
-  type = "zip"
-  excludes = [
-    "__pycache__",
-    ".mypy_cache",
-    ".pytest_cache",
-    "venv",
-  ]
+resource "aws_lambda_function" "geff_lambda" {
+  function_name = local.lambda_function_name
+  role          = aws_iam_role.geff_lambda_assume_role.arn
 
-  depends_on = [null_resource.install_python_dependencies]
-}
+  memory_size = "4096" # 4 GB
+  timeout     = "900"  # 15 mins
```

Comment:
may be worth pointing out that Snowflake currently has a 10m limit so this doesn't mislead anyone

Reply:
Is the maximum timeout for lambda function required?

Reply:
per [docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function), the default appears to be 3s

Reply:
Yes, and 15 mins is the maximum timeout for a lambda function. The best practice is to set it lower than that.. because if a lambda function needs the complete 15 mins then the function code should be optimised to reduce the chances of timeout errors.

Reply:
I can reduce the timeout or remove it to default to 3s. I believe that makes sense. Even with the async flow we'll still get back a 202 immediately and 15 mins of timeout doesn't actually make sense. 👍